src/lib/jsonParseV2/parseExpression.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+*
+* Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Fermín Galán
+*/
+
+#include <string>
+
+#include "jsonParseV2/parseExpression.h"
+
+#include "jsonParseV2/utilsParse.h"
+#include "common/errorMessages.h"
+
+
+
+/* ****************************************************************************
+*
+* parseExpression -
+*
+* This function can be called from the batch query parsing logic (subsP == NULL)
+* and from the subscription creation/update logic (subP != NULL). In the second case
+* the subscription object is filled properly
+*/
+std::string parseExpression
```

Comment:
This code was previously in subscription parse logic. I have moved to a common place, now being used but subscription parsing and batch query parsing.

NTC (informative)